src/foam/u2/wizard/StepWizardController.js

```diff
@@ -0,0 +1,321 @@
+    },
+
+    // Convenience properties
+    {
+      name: 'currentWizardlet',
+      expression: function(wizardlets, wizardPosition) {
+        var current = wizardlets[wizardPosition.wizardletIndex];
+        if ( this.visitedWizardlets.indexOf(current) == -1 ) {
+          this.visitedWizardlets.push(current);
+        }
+        return current;
+      }
+    },
+    {
+      name: 'currentSection',
+      expression: function(currentWizardlet, wizardPosition$sectionIndex) {
+        return this.currentWizardlet.currentSection = currentWizardlet
+            .sections[wizardPosition$sectionIndex];
+      }
+    },
+    {
+      name: 'isLastScreen',
+      class: 'Boolean',
+      expression: function(nextScreen) {
+        return nextScreen == null;
+      }
+    },
+    {
+      name: 'canGoBack',
+      class: 'Boolean',
+      expression: function(previousScreen) {
+        return previousScreen != null;
+      }
+    },
+    {
+      name: 'canGoNext',
+      class: 'Boolean',
+      expression: function(
+        wizardPosition, nextScreen,
+        currentSection, currentWizardlet,
+        currentSection$isValid, currentWizardlet$isValid,
+        config$allowSkipping
+      ) {
+        if ( config$allowSkipping ) return true;
+        if (
+          ! nextScreen ||
+          wizardPosition.wizardletIndex != nextScreen.wizardletIndex
+        ) {
+          if ( ! currentWizardlet$isValid ) return false;
+        }
+        return currentSection$isValid;
+      }
+    },
+    {
+      name: 'submitted',
+      class: 'Boolean'
+    }
+  ],
+
+  methods: [
+    function setupWizardletListeners(wizardlets) {
+      console.log('step wizard', this, wizardlets);
+      if ( this.wsub ) this.wsub.detach();
+      this.wsub = this.FObject.create();
+
+      var wi = 0, si = 0;
+      wizardlets.forEach((w, wizardletIndex) => {
+
+        // Bind availability listener for wizardlet availability
+        var isAvailable$ = w.isAvailable$;
+        console.log(wizardletIndex, w, w.isAvailable$, isAvailable$);
+        this.wsub.onDetach(isAvailable$.sub(() => {
+          console.log('X');
+          this.onWizardletAvailability(wizardletIndex, isAvailable$.get());
+        }));
+
+        // Bind availability listener for each wizardlet section
+        w.sections.forEach((section, sectionIndex) => {
+          var sectionPosition = this.WizardPosition.create({
+            wizardletIndex: wizardletIndex,
+            sectionIndex: sectionIndex,
+          });
+          var isAvailable$ = section.isAvailable$;
+          this.wsub.onDetach(isAvailable$.sub(() => {
+            console.log('Y');
+            this.onWizardPositionAvailabilityUpdate(
+              sectionPosition, isAvailable$.get());
+          }));
+        });
+
+      });
+
+    },
+    function detach() {
+      this.wsub.detach();
+      this.SUPER();
+    },
+    function positionAfter(pos) {
+      let subWi = pos.wizardletIndex
```

Comment:
don't think assigning functional variables for the wizardPosition helps with the readability of logic. I personally would prefer to just read the variables from where they are coming from. Especially here where we have multiple objects with similar names.
Example sections and wizardlets both have isValid.
But this is more along the lines of understanding that here in this controller we are maintaining sections, wizardlets, capabilities, ucj's, wizardletPositions and various other helper objects/properties.

Reply:
What are functional variables? I can imagine what that term would mean but it's not what I think you mean to mean.

> would prefer to just read the variables from where they are coming from

I can't really tell what you want based on that. Where are they "coming from"? I think maybe you're proposing something like `positionAfter(wi, si)` over `positionAfter(pos)`. Adding the WizardPosition model has simplified a lot of code by reducing position updates to a single slot, enabling comparison of positions, and simplifying iteration of positions.